assets/course-builder/index.js

```diff
@@ -0,0 +1,167 @@
+import apiFetch from '@wordpress/api-fetch';
+import { Button } from '@wordpress/components';
+import { select, dispatch } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { createBlock, registerBlockType } from '@wordpress/blocks';
+import { InnerBlocks, PlainText } from '@wordpress/block-editor';
+
+registerBlockType( 'sensei-lms/course-outline', {
+	title: __( 'Course Outline', 'sensei-lms' ),
+	icon: {
+		src: 'book',
+		foreground: '#32af7d',
+	},
+	category: 'layout',
+	keywords: [],
+	description: __( 'Course modules and lessons.', 'sensei-lms' ),
+	supports: {
+		align: [ 'wide', 'full' ],
+		html: false,
+	},
+	attributes: {
+		_version: {
+			type: 'int',
+			default: 0,
+		},
+	},
+
+	edit( props ) {
+		return <CourseOutlineEditorBlock { ...props } />;
+	},
+
+	save( { innerBlocks } ) {
+		const courseId = select( 'core/editor' ).getCurrentPostId();
+		if ( courseId ) {
+			apiFetch( {
```

Comment:
For all POCs, I believe this is not the best approach to save this data. Since it wasn't the main idea of these POCs, I didn't dig into that for now. So when we implement a solution, it'd worth investigate it better.